initial schema docu (#32)
nektos/act-docs: src/usage/schema.md

# Schema

> Did you land here with cryptic errors? Then please install the following official GitHub Actions VSCode Extension https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions and review all errors/warnings shown there.

Reviewer: ```suggestion
Did you land here because of cryptic errors? Then please install the following official GitHub Actions VSCode Extension https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions and review all errors/warnings shown by it.

```

> Once you are shure this is an error in act, please look for existing issues or report a new issue here https://github.com/nektos/act/issues.

Reviewer: ```suggestion
Once you are sure this is an error in act, please look for existing issues or report a new issue here https://github.com/nektos/act/issues.
```
I've been meaning to make this correction in many places

Reviewer: I don't want to mix things, but there's an unrelated item with `here` links (your text is actually considerably better than the average)
and trailing punctuation (GitHub screws this up, which means I get to file another bug):
<img width="369" alt="image" src="https://github.com/user-attachments/assets/492d46cc-37d9-4fa4-bd94-8993d965d855">
Specifically [ https://github.com/nektos/act/issues.](https://github.com/nektos/act/issues.) is a `404`, and if GitHub can mess this up, you can be sure that some terminals will too (I've seen a number do it).
```suggestion
Once you are sure this is an error in act, please look for existing issues or visit https://github.com/nektos/act/issues/new/choose to report a new issue.
```

> _Act's builtin schema validation errors are not easy to understand while keeping the development effort reasonable for open source_

Reviewer: ```suggestion
_Act's built-in schema validation errors are not easy to understand while keeping the development effort reasonable for open source_

```

# Schema

Did you land here because of cryptic errors? Then please install the following official GitHub Actions VSCode Extension https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions and review all errors/warnings shown by it.

Once you are sure this is an error in act, please look for existing issues or visit https://github.com/nektos/act/issues/new/choose to report a new issue.

> _Act's built-in schema validation errors are not easy to understand while keeping the development effort reasonable for open source_

Reviewer: ```suggestion
_Act's built-in schema validation errors are not easy to understand while keeping the development effort reasonable for open source._
```

I'm not entirely certain what this sentence is trying to say, but I'm assuming it's a sentence.

Author: I'm in my own bubble :), not everything I write makes sense.

Sometimes not getting reviews for months can lead to such statements.

Actually actions/runner is worse, I gave it up and just do everything I want in my own fork.

Reviewer: I've tried contributing to a bunch of the actions/* repositories, a number of them are indeed wastelands.
